lib/apiservers/engine/backends/container.go

```diff
@@ -1605,6 +1599,70 @@ func (c *Container) containerAttach(name string, ca *backend.ContainerAttachConf
 	return nil
 }
 
+// ContainerRename changes the name of a container, using the oldName
+// to find the container. An error is returned if newName is already
+// reserved.
+func (c *Container) ContainerRename(oldName, newName string) error {
+	defer trace.End(trace.Begin(newName))
+
+	if oldName == "" || newName == "" {
+		err := fmt.Errorf("neither old nor new names may be empty")
+		log.Errorf("%s", err.Error())
+		return err
```

Comment:
The error message is good - but we should return a docker API error:

```go
err := fmt.Errorf("neither old nor new names may be empty")
return derr.NewErrorWithStatusCode(err, http.StatusInternalServerError)
```

Since Docker only returns 404, 409 or 500 errors for `rename`, I suppose 500 is suitable here.